tests/thread_flood/main.c

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (C) 2015 Hamburg University of Applied Sciences
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup tests
+ * @{
+ *
+ * @file
+ * @brief Thread flooding test.
+ *
+ * Spawns sleeping threads till the scheduler's capacity is exhausted.
+ *
+ * @author  Andreas "Paul" Pauli <andreas.pauli@haw-hamburg.de>
+ *
+ * @}
+ */
+
+#include <errno.h>
+#include <stdio.h>
+#include "thread.h"
+#include "lpm.h"
+
+char dummy_stack[THREAD_STACKSIZE_DEFAULT];
+
+/* One stack for all threads. DON'T TRY THIS AT HOME!! */
+void *thread_func(void *arg)
+{
+    return arg;
+}
+
+int main(void)
+{
+    kernel_pid_t thr_id = KERNEL_PID_UNDEF;
+    puts("Start spawning\n");
+    while (-EOVERFLOW != thr_id) {
+        thr_id = thread_create(
+                     dummy_stack, sizeof(dummy_stack),
+                     THREAD_PRIORITY_MAIN - 1, CREATE_SLEEPING | CREATE_STACKTEST,
+                     thread_func, NULL, "dummy");
+    }
+    if (-EOVERFLOW == thr_id) {
+        puts("Thread creation successful aborted\n");
+    }
+    lpm_set(LPM_OFF);
```

Comment:
The original idea in the unittests application was primarily to keep an automatic test from being stuck in the idle thread. However, since then I observed that just shutting down the node leads to some problems with the output (mainly uncompleted lines) on physical boards. A `pexlect` script or similar tool could deal with this. See #3309 for the `tests/unittests` application.